### PR TITLE
Support Spark 1.5.2

### PR DESCRIPTION
**Pull request by vidma**
- simply copied over the `modules/spark/src/main/scala_2.10/spark-1.5.1` into `modules/spark/src/main/scala_2.10/spark-1.5.2`. works fine.
- rc2 should work by simply adding repo url into resolvers.
- add a note, on how to workaround `hive+parquet` to work in `spark 1.5.2 with CDH 5.4.4+`

@jarutis @andypetrella

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -79,6 +79,7 @@ object Dependencies {
     ExclusionRule("org.apache.ivy", "ivy"),
     ExclusionRule("javax.servlet", "servlet-api"),
     ExclusionRule("org.mortbay.jetty", "servlet-api"),
+    // P.S. Need to comment this out for spark 1.5.2 work with Hive+parquet in CDH 5.4.4+ (hadoop 2.6.0)    
     ExclusionRule("com.twitter", "parquet-hadoop-bundle")
```
Comment (vidma) [inline project/Dependencies.scala]: a fancy if might be better, but don't want to tinker with this. commenting it out makes it work for us.

the issue seem to be the fact that `parquet-format & parquet` moved their namespaces from `parquet.*` into `org.apache.parquet` recently. spark 1.5, uses the new parquet.

hive libs (likely the ones bundled into `spark-notebook` ? ) on the over hand, seem to use the old lib-path.
I don't think that we include any CDH hive libs (except for hadoop natives).

File: build.sbt [hunk 2]
```diff
@@ -63,7 +63,8 @@ resolvers in ThisBuild ++= Seq(
   // docker
   "softprops-maven" at "http://dl.bintray.com/content/softprops/maven",
   //spark cutting edge
-  "spark 1.5.0-rc2" at "https://repository.apache.org/content/repositories/orgapachespark-1141"
+  "spark 1.5.0-rc2" at "https://repository.apache.org/content/repositories/orgapachespark-1141",
+  "spark 1.5.2-rc1" at "https://repository.apache.org/content/repositories/orgapachespark-1151"
```
Comment (jarutis) [inline build.sbt]: as spark 1.5.2 is out I doubt we still need this

**Comment by andypetrella** [comment]
sounds really good man!

I'm thinking at two things:
* it might be worth now having another heuristic for these spark-x.y folders, maybe better to have a `default` folder which is the one to use if the spark pattern doesn't pass. And when there are changes in the REPL, then you rename it to the latest version that worked

Then another parameter should be added to add a repo (like for rc-X) to the list.

This would be then super easy to add support for current dev versions for instance.

* the second thought is way easier, in order to plug this heuristic regarding parquet, would be ok creating a small issue describing shortly the problem? Then we can refer it in the PR?

Anyway thanks for the great work!

**Comment by vidma** [comment]
merging this so #381 can be rebased more nicely.

**Pull request merged**